elastic/eui
# Build targeting Jest with relevant transforms

### Summary

Adds a third build target: commonjs via babel with async function and dynamic import transforms applied.
The goal is to eliminate needing to transform EUI from within Kibana's Jest setup, using `moduleNameMapper` to point to the targeted build in testing contexts. [See the Draft PR](https://github.com/elastic/kibana/pull/55877) for changes to be make to Kibana.

Resolves #2065

This also proposes a pattern for overriding components ad hoc for the benefit of testing. See changes related to `icons.tsx`, in which the component is fully rewritten to avoid dynamic imports. Looking for opinions on what we should actually render when it comes to icon snapshots.

Testing steps:
1) `yarn build` from eui
2) Either `npm pack` and change Kibana's `package.json`, or copy over a subset of directories to Kibana `node_modules`
3) `yarn test:jest` from kibana
4) Expected output is a slew of test failures related to icon snapshots

### Checklist

~~- [ ] Check against **all themes** for compatability in both light and dark modes~~
~~- [ ] Checked in **mobile**~~
~~- [ ] Checked in **IE11** and **Firefox**~~
~~- [ ] Props have proper **autodocs**~~

- [x] Added **documentation** examples (_**document creating `*.testenv.*` files for testing purposes**_)
- [x] Added or updated **jest tests**

~~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~~

- [x] A [changelog](https://github.com/elastic/eui/blob/master/CHANGELOG.md) entry exists and is marked appropriately
